packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -629,24 +629,48 @@ public boolean onActivityResult(
     return true;
   }
 
-  private void handleChooseImageResult(int resultCode, Intent data) {
-    if (resultCode == Activity.RESULT_OK && data != null) {
-      Uri uri = data.getData();
-      // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could be null.
-      if (uri == null) {
-        ClipData clipData = data.getClipData();
-        if (clipData != null && clipData.getItemCount() == 1) {
-          uri = clipData.getItemAt(0).getUri();
+  private ArrayList<MediaPath> getPathsFromIntent(Intent data, boolean includeMimeType) {
+    ArrayList<MediaPath> paths = new ArrayList<>();
+
+    Uri uri = data.getData();
+    // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could
+    // be null.
+    if (uri == null) {
+      ClipData clipData = data.getClipData();
+
+      // If data.getData() and data.getClipData() are both null, we are in an error state. By
+      // convention we return null from here, and then finish with an error from the corresponding
+      // handler.
+      if (clipData == null) {
+        return null;
+      }
+
+      for (int i = 0; i < data.getClipData().getItemCount(); i++) {
+        uri = data.getClipData().getItemAt(i).getUri();
+        // Same error state as above.
+        if (uri == null) {
+          return null;
```

Comment:
I don't think this should return null here. what if there are multiple items inside of the ClipData, and the first few have valid Uri, but at a certain index it doesn't? In that case, the paths variable should already have values inside it

Reply:
I was on the fence because of that case - but at the same time I wouldn't want to silently return just the strict subset of images that we successfully got uris for. 

cc @tarrinneal as the overall owner of the plugin, do you have any preferences here on the behavior of the plugin for the case of failing to get the uri for one of the images in a multi-image choice action? I think currently it would crash if that were to happen

Reply:
I think it would make more sense to throw and handle errors across these methods. Perhaps @stuartmorgan would also like to weigh in.

Reply:
I would also vote for throwing an error in that case rather than silently failing (or silently dropping some paths).